src/NinjaTools.FluentMockServer.API/Extensions/HttpRequestExtensions.cs

```diff
@@ -0,0 +1,39 @@
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Http;
+
+namespace NinjaTools.FluentMockServer.API.Extensions
+{
+    public static class HttpRequestExtensions
+    {
+
+        /// <summary>
+        /// Retrieve the raw body as a string from the Request.Body stream
+        /// </summary>
+        /// <param name="request">Request instance to apply to</param>
+        /// <param name="encoding">Optional - Encoding, defaults to UTF8</param>
+        /// <returns></returns>
+        public static async Task<string> GetRawBodyStringAsync([NotNull] this HttpRequest request, Encoding encoding = null)
+        {
+            if (encoding == null)
+                encoding = Encoding.UTF8;
+
+            using var reader = new StreamReader(request.Body, encoding);
+            return await reader.ReadToEndAsync();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/app/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4722826)